railties/lib/rails/test_unit/test_requirer.rb

```diff
@@ -15,7 +15,7 @@ def require_files(patterns)
       private
         def expand_patterns(patterns)
           patterns.map do |arg|
-            arg = arg.gsub(/:(\d+)?$/, '')
+            arg = arg.gsub(/(:\d*)+?$/, '')
```

Comment:
@prathamesh-sonpatki I just checked now. If we change regex to `(:\d+)+`  then the `test_line_filter_without_line_runs_all_tests` test is failing. Do we need to consider this case now?